lib/question.sql

```diff
@@ -41,3 +41,48 @@ ORDER BY
 SELECT i.*
 FROM issues AS i
 WHERE i.variant_id = $variant_id;
+
+-- BLOCK select_submission_info
+SELECT
+    to_jsonb(gj) AS grading_job,
+    to_jsonb(s) AS submission,
+    to_jsonb(v) AS variant,
+    to_jsonb(iq) AS instance_question,
+    to_jsonb(q) AS question,
+    to_jsonb(aq) AS assessment_question,
+    to_jsonb(ai) AS assessment_instance,
+    to_jsonb(a) AS assessment,
+    to_jsonb(c) AS course,
+    gj.id AS grading_job_id,
+    grading_job_status(gj.id) AS grading_job_status,
+    format_date_full_compact(s.date, coalesce(ci.display_timezone, c.display_timezone)) AS formatted_date,
+    (
+        SELECT count(*)
+        FROM submissions AS s2
+        WHERE s2.variant_id = s.variant_id
+        AND s2.date < s.date
+    ) + 1 AS submission_index,
+    (
+        SELECT count(*)
+        FROM submissions AS s2
+        WHERE s2.variant_id = s.variant_id
+    ) AS submission_count
+FROM
+    grading_jobs AS gj
+    LEFT JOIN submissions AS s ON (s.id = gj.submission_id)
+    LEFT JOIN variants AS v ON (v.id = s.variant_id)
```

Comment:
I'd prefer to only write LEFT JOIN when it's possible that a JOIN might fail. So `submissions`, `variants`, `questions`, `pl_courses` can all be plain JOINs.